src/main/java/eu/adlogix/appnexus/oas/client/util/Credentials.java

```diff
@@ -0,0 +1,111 @@
+package eu.adlogix.appnexus.oas.client.util;
+
+import static eu.adlogix.appnexus.oas.client.resources.ResourceUtils.loadResourceInputStream;
+
+import java.io.FileInputStream;
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.Properties;
+
+import org.apache.commons.lang3.StringUtils;
+
+import eu.adlogix.appnexus.oas.client.domain.PushLevel;
+
+
+public final class Credentials {
```

Comment:
Not sure though, but you should check 'spring boot'. I think they have clean ways for external parameters and files